[git-webkit] Automatically CC radar when creating bugs

JonWBedard opened the pull request:
#### 291e6f5f407d5dd7e26946791bad5f95cf668eb1
<pre>
[git-webkit] Automatically CC radar when creating bugs
<a href="https://bugs.webkit.org/show_bug.cgi?id=244109">https://bugs.webkit.org/show_bug.cgi?id=244109</a>
&lt;rdar://problem/98857423&gt;

Reviewed by Dewei Zhu.

* Tools/Scripts/libraries/webkitbugspy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
(Tracker.Encoder.default): Encode Radar importer.
(Tracker.__init__): Add Radar importer.
(Tracker.populate): Prefer Radar importer bugs to ones mentioned in the description.
(Tracker.cc_radar): CC the Radar importer.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
(Tracker.cc_radar): Add.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py:
(Issue.cc_radar): Add.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/base.py:
(Base.add): Return issue ID.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py:
(Bugzilla._issue): Handle CC addition and add comment if the Radar importer is CCed.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py: Add Radar importer.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py:
(Tracker.cc_radar): Add.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/bugzilla_unittest.py:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tracker.py:
(Tracker.from_json): Handle radar_importer object.
(Tracker.cc_radar): Add.
* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/__init__.py: Add webkitbugspy logger.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
(Branch.main): CC radar importer if user has radarclient.
* metadata/trackers.json: Add radar importer.

Canonical link: <a href="https://commits.webkit.org/254281@main">https://commits.webkit.org/254281@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/52e1d10436c66df0a20782361e371d6e7a9fcd3c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88617 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33179 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19504 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/97810 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/154341 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/92613 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/31671 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27260 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/80842 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92446 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94247 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25126 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75569 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25063 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/92209 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80010 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/80078 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68026 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/29356 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/14077 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/87677 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29201 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15090 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3022 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/32632 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/38022 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31326 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34199 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

alancoon reviewed (approved):
I don't think I have review privileges, but looks good to me.

dewei-zhu commented on metadata/trackers.json:
```diff
@@ -8,6 +8,10 @@
         ],
         "redact" : {
             "product:Security": true
+        }, "radar_importer": {
```
Nit: new line for "radar_importer" and align with "redact" indentation?

dewei-zhu commented on Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
```diff
@@ -447,3 +466,52 @@ def create(
+            time.sleep(10)
+            issue._comments = None
+            issue._references = None
+
```
Should we write some error message when we exhaust timeout?

JonWBedard replied:
Good idea, added that.

dewei-zhu commented on Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
```diff
@@ -447,3 +466,52 @@ def create(
+                if isinstance(reference.tracker, RadarTracker):
+                    return reference
+            if not block or not did_modify_cc:
+                break
```
Direct `return None`? Also might worth some logging?

JonWBedard replied:
I don't think we need logging in this case, because the caller didn't ask us to block on the radar importer or we didn't need to modify the CC list (usually because the radar importer was already CCed)

dewei-zhu commented on Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
```diff
@@ -447,3 +466,52 @@ def create(
+
+        did_modify_cc = False
+        if self.radar_importer not in issue.watchers:
+            log.info('CCing {}'.format(self.radar_importer.name))
```
Could you explain why we want to mix usages between `sys.stderr` and `log`?

JonWBedard replied:
`git-webkit` respects `-v` and `-q`. I try and reserve `sys.stderr` for errors, and `log` for informational messages. Put another way, you can silence `log` messages (or they may be silenced by default, as is the case with `log.info`) where as you cannot silence `sys.stderr` or `sys.stdout` with command line arguments.

dewei-zhu replied:
I see.

dewei-zhu reviewed (commented):
Have one question would like to be answered before r+.

dewei-zhu reviewed (approved):
r=me

JonWBedard committed:
[git-webkit] Automatically CC radar when creating bugs
https://bugs.webkit.org/show_bug.cgi?id=244109
<rdar://problem/98857423>

Reviewed by Dewei Zhu.

* Tools/Scripts/libraries/webkitbugspy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
(Tracker.Encoder.default): Encode Radar importer.
(Tracker.__init__): Add Radar importer.
(Tracker.populate): Prefer Radar importer bugs to ones mentioned in the description.
(Tracker.cc_radar): CC the Radar importer.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
(Tracker.cc_radar): Add.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py:
(Issue.cc_radar): Add.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/base.py:
(Base.add): Return issue ID.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py:
(Bugzilla._issue): Handle CC addition and add comment if the Radar importer is CCed.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py: Add Radar importer.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py:
(Tracker.cc_radar): Add.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/bugzilla_unittest.py:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tracker.py:
(Tracker.from_json): Handle radar_importer object.
(Tracker.cc_radar): Add.
* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/__init__.py: Add webkitbugspy logger.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
(Branch.main): CC radar importer if user has radarclient.
* metadata/trackers.json: Add radar importer.

Canonical link: https://commits.webkit.org/254281@main
